app/Http/Controllers/AlbumController.php

```diff
@@ -52,4 +53,36 @@ public function store(Request $request)
             ->route('albums.index')
             ->with('success', "Successfully created {$artist->name} - {$request->input('title')}");
     }
+
+    public function edit($id)
+    {
+        $album = DB::table('albums')->where('id', '=', $id)->first();
+        $artists = DB::table('artists')->orderBy('name')->get();
+
+        return view('album.edit', [
+            'album' => $album,
+            'artists' => $artists,
+        ]);
+    }
+
+    public function update($id, Request $request)
+    {
+        $request->validate([
+            'title' => 'required|max:20',
+            'artist' => 'required|exists:artists,id',
+        ]);
+
+        DB::table('albums')->where('id', '=', $id)->update([
+            'title' => $request->input('title'),
+            'artist_id' => $request->input('artist'),
+        ]);
+
+        $artist = DB::table('artists')
+            ->where('id', '=', $request->input('artist'))
+            ->first();
+
+        return redirect()
+            ->route('album.edit', [ 'id' => $id ])
+            ->with('success', "Successfully updated {$artist->name} - {$request->input('title')}");
+    }
```

Comment:
This method is very similar to the `store()` method but instead of inserting a record, it updates a record and the flash message is different.